src/modules/common/hooks/useAuth.ts

```diff
@@ -0,0 +1,41 @@
+import { useUserStore } from "@/modules/common/hooks/useUserStore";
+import {
+  getUser,
+  login,
+  register,
+  type LoginBodyParam,
+} from "@/modules/services/auth";
+import { useMutation, useQuery } from "@tanstack/react-query";
+
+export const useLogin = () => {
+  const { refetch } = useUser();
+  return useMutation({
+    mutationFn: ({ username, password }: LoginBodyParam) =>
+      login({ username, password }),
+    onSuccess: () => {
+      void refetch();
+    },
+  });
+};
+
+export const useRegister = () => {
+  return useMutation({
+    mutationFn: ({ username, password }: LoginBodyParam) =>
+      register({ username, password }),
+  });
+};
+
```

Comment:
use RegisterBodyParam instead of LoginBodyParam Type

Reply:
✅